### PR TITLE
Use /ping endpoint to check connection to API

### PR DESCRIPTION
**Pull request by rahul-thakoor**
Change-type: minor

### CODE REVIEW

File: src/stage1/migrate_info/balena_cfg_json.rs [hunk 1]
```diff
@@ -230,3 +198,33 @@ impl BalenaCfgJson {
         &self.file
     }
 }
+
+fn check_api(api_endpoint: &str) -> Result<()> {
+    let res = Client::builder()
+        .build()
+        .upstream_with_context("Failed to create https client")?
+        .get(api_endpoint)
```
Comment (lmbarros) [inline src/stage1/migrate_info/balena_cfg_json.rs:206]: Isn't this URL coming directly from `config.json`? So, unless I missed something you'd need to append `/ping` here, right?

**Review by lmbarros — CHANGES_REQUESTED** [review]

File: src/stage1/migrate_info/balena_cfg_json.rs [hunk 2]
```diff
@@ -230,3 +198,33 @@ impl BalenaCfgJson {
...
+    if status.is_success() && response.trim() == "OK" {
+        info!("connection to api: {} is ok", &api_endpoint);
+        Ok(())
+    } else {
+        return Err(Error::with_context(
+            ErrorKind::InvState,
+            &format!(
+                "failed to connect to api server @ {}. Your device might not come online",
+                &api_endpoint
+            ),
```
Comment (lmbarros) [inline src/stage1/migrate_info/balena_cfg_json.rs:227]: If we reach this point we have already connected to the API server. The problem here is that it is unresponsive (or failing to reply to the ping, at least).

Suggestion (not tried on a compiler :-) ):

```suggestion
            &format!(
                "got an unexpected reply from the API server @ {}: '{}'",
                &api_endpoint,
                &response
            ),
```

File: src/stage1/migrate_info/balena_cfg_json.rs (hunk 1 above)
Comment (rahul-thakoor) [reply src/stage1/migrate_info/balena_cfg_json.rs]: nice catch! i had /ping in the test URL 😓

File: src/stage1/migrate_info/balena_cfg_json.rs [hunk 3]
```diff
@@ -230,3 +195,34 @@ impl BalenaCfgJson {
...
+fn check_api(api_endpoint: &str) -> Result<()> {
+    let ping_endpoint = format!("{}/ping", api_endpoint);
+    let res = Client::builder()
+        .build()
+        .upstream_with_context("Failed to create https client")?
+        .get(&ping_endpoint)
+        .send()
+        .upstream_with_context(&format!(
+            "Failed to send https request url: {}",
+            &api_endpoint
```
Comment (lmbarros) [inline src/stage1/migrate_info/balena_cfg_json.rs:208]: I think using `ping_endpoint` here would be a little, super tiny bit more informative:

```suggestion
            "Failed to send https request url: {}",
            &ping_endpoint
```

**Pull request merged**